src/main/java/com/scottlogic/AuthorPostSorter.java

```diff
@@ -0,0 +1,16 @@
+package com.scottlogic;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+public class AuthorPostSorter implements PostSorter {
+
+    @Override
+    public List<UserPost> sort(List<UserPost> inputList) {
+        if(inputList != null){
+                inputList.sort(Comparator.comparing(UserPost::getAuthor));
```

Comment:
I missed this on first pass, but this is mutating the `inputList` before returning it. In general you shouldn't mutate input values as when this causes bugs they are really hard to debug

Reply:
Yeah will be sure to change that